File: dotnet/DesignPatterns/CreationalDesignPatterns/BuilderDesignPattern/BuilderDesignPattern.Example1/ReportBuilder.cs
namespace BuilderDesignPattern.Example1;

public class ReportBuilder : IReportBuilder
{
    private readonly Report _report = new();

    public IReportBuilder SetTitle(string title)
    {
        _report.Title = title;
        return this;
    }

    public IReportBuilder SetContent(string content)
    {
        _report.Content = content;
        return this;
    }

    public IReportBuilder SetHeader(string header)
    {
        _report.Header = header;
        return this;
    }

    public IReportBuilder SetFooter(string footer)
    {
        _report.Footer = footer;
        return this;
    }

    public IReportBuilder SetFont(string font)
    {
        _report.Font = font;
        return this;
    }

    public Report Build()
    {
        return _report;
    }
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/BuilderDesignPattern/BuilderDesignPattern.Example1/src/ReportBuilder.ts
import { IReportBuilder } from './IReportBuilder';
import { Report } from './Report';

export class ReportBuilder implements IReportBuilder {
  private readonly _report = new Report();

  setTitle(title: string): IReportBuilder {
    this._report.title = title;
    return this;
  }

  setContent(content: string): IReportBuilder {
    this._report.content = content;
    return this;
  }

  setHeader(header: string): IReportBuilder {
    this._report.header = header;
    return this;
  }

  setFooter(footer: string): IReportBuilder {
    this._report.footer = footer;
    return this;
  }

  setFont(font: string): IReportBuilder {
    this._report.font = font;
    return this;
  }

  build(): Report {
    return this._report;
  }
}
